### ISSUE TITLE
cat: /sys/devices/virtual/dmi/id/product_name: No such file or directory

### ISSUE THREAD

**Issue by THEWHITEBOY503** [THEWHITEBOY503/bash-showoff-script#3]
When running confetch from WSL (Windows Subsystem for Linux), attempting to automatically grab the model identifier from `/sys/devices/virtual/dmi/id/product_name` results in the error:
`cat: /sys/devices/virtual/dmi/id/product_name: No such file or directory`

**Comment by THEWHITEBOY503** [comment]
Here's how I'm working around this issue:
new code:
```bash
if [ "$MODEL" = true ] ; then
    if ! cat /sys/devices/virtual/dmi/id/product_name 2>/dev/null; then
        if grep -q -i "Microsoft" /proc/sys/kernel/osrelease || grep -q -i "WSL" /proc/sys/kernel/osrelease; then
            tMODEL="Windows Subsystem for Linux"
        else
            # Failed to grab the model identifier. (Probably running on WSL). 
            tMODEL="Computer"
        EF1="Warning: Model grab failed. Defaulting to "Computer"; Set MODEL manually in config.txt to dismiss this message."
        fi
    else
        tMODEL=$(cat /sys/devices/virtual/dmi/id/product_name)
    fi
else
    tMODEL=$MODEL
fi
```

Now, the script first checks to see if it can't cat the file.  If it fails to cat the file, it first checks to see if the user is running on WSL. If they are, it sets tMODEL to "Windows Subsystem for Linux", otherwise it defaults to setting tMODEL = "Computer". It then flags EF1 (Error Flag 1) with the warning message:
`Warning: Model grab failed. Defaulting to "Computer"; Set MODEL manually in config.txt to dismiss this message.`

**Comment by THEWHITEBOY503** [comment]
Fixed in [fd0abea](https://github.com/THEWHITEBOY503/confetch/commit/fd0abea33ffd75bbe3771b1e66e600604d944b48).
